src/main/java/bio/terra/service/dataset/dao/DatasetDao.java

```diff
@@ -385,11 +400,13 @@ public Dataset retrieve(UUID id) {
      * @param id the dataset id
      * @return the DatasetSummary object
      */
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
```

Comment:
Would it make sense to have `@Transaction(propagation = Propagation.REQUIRED)` as a class level annotation as the default transactionality mode and then we can override as needed in sub methods? Makes us err on the side of having all public methods in the dao being transactional

Reply:
This would be interesting! A sort of `strict` for transactions.